boilerplate/update

```diff
@@ -10,6 +10,9 @@
 # script.
 
 set -e
+if [ "$BOILERPLATE_SET_X" ]; then
```

Comment:
I would recommend calling this `DEBUG` or `BP_DEBUG` and using this as a convention in all scripts.

Reply:
> and using this as a convention in all scripts.

To be clear: By putting it in the two places I've put it, it automatically applies to any `make update_boilerplate` or `make test` all the way through. We don't need to add it to individual test cases or anything.

> I would recommend calling this `DEBUG` or `BP_DEBUG`

I can rename it if you like.

Reply:
I meant if there are any other shell scripts if they all use the same environment variable for debug logs then it would be quite convenient. Because the `-x` flag won't be passed to any shell scripts which are called from this script. But we can discuss that later separately.